### ISSUE TITLE
Difference between Trans and PairExcitation file

### ISSUE THREAD

**Issue by SuHang980715** [issp-center-dev/HPhi#146]
Hi, when I read the guide file of HPhi, I found that there were two files Trans (sec. 4.5) and Pair Excitation (sec. 4.17). But both of them included the terms c_{i, \sigma_1} c_{j, \sigma_2}^{\dagger} or its Hermition conjugate terms. So could I ask about the difference between them? Thanks!

**Comment by tmisawa** [comment]
`Trans` is used for specifying the hopping parameters in Hamiltonians.
`PairExcitation` is used for specifying **the pair excitations**, i.e., two-body dynamical Green functions. 
For details of the pair excitations and the dynamical Green functions, please see sec. 5.4.  
The form of the pair excitations is defined in Eq. 5.18.

http://issp-center-dev.github.io/HPhi/manual/master/en/html/algorithm/DynamicalGreen_en.html

**Comment by k-yoshimi** [comment]
In the expert mode, the spectrum function can be calculated by following steps.
1. Calculate the ground (target) state by running HPhi (Please set ``OutputEigenVec 1 `` in namelist.def ).
2. Define excitation operators in ``PairExcitation``/``SingleExcitation`` files.
3. Add input file name of ``PairExcitation``/``SingleExcitation`` and ``SpectrumVec``  in namelist.def.
    Here, the file name specified by ``SpectrumVec`` is the output file of the target eigen state obtained by step 1 which corresponds to $\Phi_0$ in  http://issp-center-dev.github.io/HPhi/manual/master/en/html/algorithm/DynamicalGreen_en.html. 
4. Set ``CalcSpec 1`` in "calcmod.def".
5.  Define frequencies (corresponding to z in eq.5.16 in http://issp-center-dev.github.io/HPhi/manual/master/en/html/algorithm/DynamicalGreen_en.html) by setting ``OmegaMin`` , ``OmegaMax``, ``NOmega``, ``OmegaIm`` in "modpara.def".
7. Calculate spectrum function by running HPhi.

```
Memo: List of Related files
- namelist.def
   - PairExcitation, SingleExcitation, SpectrumVec 
- Input file specified by CalcMod in namelist.def (ex. calcmod.def)
  - OutputEigenVec, CalcSpec
- Input file specified by ModPara in namelist.def (ex. modpara.def)
  - OmegaMin, OmegaMax, NOmega, OmegaIm
```


>Mainly, my aim is to get suspectibility for a spin model, and maybe such a dynamical Green Function could give me some information. But I'm not sure if it is in imaginary-time representation. Could you give some suggestions about this by using HPhi?

In HPhi, both real and Matsubara frequency representation can be treated by setting `OmegaMin`` , ``OmegaMax``, ``NOmega``, ``OmegaIm`` in "modpara.def". Please see the manual ([here](http://issp-center-dev.github.io/HPhi/manual/master/en/html/filespecification/expertmode_en/ModPara_file_en.html#calculating-dynamical-greens-functions)).

I think that checking input files for expert mode and output files by using standard mode is helpful. 
The tutorial using standard mode is written [here](https://issp-center-dev.github.io/HPhi/manual/develop/tutorial/en/html/dynamical/index.html).

**Comment by SuHang980715** [comment]
Thanks for your response, which is very helpful for me. And for pairexcitation part, I have a naive question about that: should I add some coefficients when defining it? I mean, could I only define the coefficient as 1 for all terms? Thanks for your nice again.

**Comment by k-yoshimi** [comment]
As described [here](http://issp-center-dev.github.io/HPhi/manual/master/en/html/filespecification/expertmode_en/PairExcitation_file_en.html), you can set coefficients as you do. In this example, for simplicity, 1 is set to all excitation operators.

**Comment by k-yoshimi** [comment]
Yes, the example is the case for T=0. If you want to calculate dynamical Green functions at finite temperature exactly, all eigen vectors and values are needed. However, as an approximation, with increasing the number of eigenstates, the  dynamic Green's functions approach to exact ones. If $\beta(\epsilon_n -\mu)$ is sufficiently larger than 1, the Boltzmann factor becomes almost 0. Then,  you can obtain almost exact dynamical Green functions.

**Comment by SuHang980715** [comment]
Thank for your nice help. And for static susceptibility, is this possible to be calculated by HPhi? (at finite temperature), because I'm not sure about this.

**Comment by SuHang980715** [comment]
And could I ask about that: are there any ways to get all of eigenvectors in expert mode? Because I think it is enough to me, to some extent.

**Comment by tmisawa** [comment]
By specifying `CalcType=2` and `OutputEigenVec=1` in `calcmod.def`, 
you can do full diagonalization and obtain all eigenvectors.

http://issp-center-dev.github.io/HPhi/manual/master/en/html/filespecification/expertmode_en/CalcMod_file_en.html

**Comment by SuHang980715** [comment]
Sorry for late reply, I have set them, however, I still did not obtain all eigenvectors. 
This is my calcmod.def file: 

CalcType   2
CalcModel   4
ReStart   0
CalcSpec   0
CalcEigenVec   0
InitialVecType   0
InputEigenVec   0
OutputEigenVec   1
InputHam   0
OutputHam   0
OutputExVec   0

And, this is my namelist.def file

         ModPara  modpara.def
         LocSpin  locspn.def
           Ising  ising.def
           Trans  trans.def
        OneBodyG  greenone.def
        TwoBodyG  greentwo.def
         CalcMod  calcmod.def
  PairExcitation  pair.def
     SpectrumVec  zvo_eigenvec_0

I use this command to run HPhi: 

HPhi -e namelist.def

How should I do to obtain all eigenvectors? Could you give more suggestions about this? Thanks very much~~

**Comment by tmisawa** [comment]
Closing this issue as completed.

The original questions about the roles of `Trans` and `PairExcitation`, and the expert-mode spectrum workflow, were answered in this thread. The later request to output FullDiag eigenvectors is now tracked separately in #214.

This closure is recorded in #289.
